### PR TITLE
Feature/forms

### LINKED ISSUE #16: Sign-in && Sign-up Page
\***\*Task Description\*\***
Build  Sign-in && Sign-up page

\***\*Technical Considerations\*\***
Using React.js ( with one of the libraries/ frameworks)

\***\*Additional Considerations\*\***

### PR DESCRIPTION
**Pull request by olivcamj**
Added form and form input components (for reusability) and used them to create signin and signup forms.

## Images
Screen Shot 2021-07-31 at 8 35 06 PM

<details>
<summary>Tablet </summary>
<br>
Screen Shot 2021-07-31 at 8 35 29 PM
</details>

<details>
<summary>Desktop</summary>
<br>

</details>
<details>
<summary>Empty form</summary>
<br>
Screen Shot 2021-07-31 at 7 35 19 PM
</details>

<details>
<summary>Filled out and passing client-side validation</summary>
<br>
Screen Shot 2021-07-31 at 7 34 30 PM
</details>

<details>
<summary>Incomplete form failing client-side validation</summary>
<br>
Screen Shot 2021-07-31 at 8 12 22 PM

</details>

<details>
<summary>Added favicon </summary>
<br>
Screen Shot 2021-07-31 at 7 53 57 PM

- currently only working(appearing) on Chrome
- ` <link rel="icon" type="image/png" href="../favicon.png" />`
</details>

Resolves: #16

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  MINLENGTH,
+  validEmailRegexp,
+  validPasswordRegexp,
+} from '../../utils/constants';
+
+function App() {
+  const [isBtnDisabled, setIsBtnDisabled] = useState(true);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
```
Comment (ArturoGrullon) [inline src/components/App/App.js:17]: One little suggestion here, 
- name
- email
- password

Could be saved in one useState just as errors, something like: 
`const [userData, setUserData] = useState({name: '', email: '', password: ''})`

File: src/components/App/App.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  };
+
+  const handlePassword = (e) => {
+    setPassword(e.target.value);
+  };
+
+  const handleName = (e) => {
+    setName(e.target.value);
+  };
+
```
Comment (ArturoGrullon) [inline src/components/App/App.js:58]: So here you could use just one function, something like: 
`  const handleUserData = (value, keyName) => {
    setUserData((state) => ({
      ...state,
      [keyName]: value,
    }));
  };`

**Review by ArturoGrullon — APPROVED** [review]
Overall seems pretty organized, and readable. Nice job Olivia 👍

No objections from me, but maybe change the useEffects for a function that makes the same functionality, the only difference is that will only run when the user "clicks" continue or something. Also, there are conflicts to be resolve maybe take a look at that

**Review by ryanbest99 — APPROVED** [review]

**Review by sccala — APPROVED** [review]
Thank you for your hard work.

**Pull request merged**